actors/organizer.go

```diff
@@ -314,79 +400,310 @@ func (o *Organizer) handleCreatePoll(msg message.Message, canal string, query me
 		Extra:    data.Extra,
 	}
 
-	errs = db.CreateChannel(event, o.database)
+	errs = db.CreateChannel(poll, o.database)
+	if errs != nil {
+		log.Printf("An error occured, unable to create channel in the database")
+		return nil, errs
+	}
+
+	errs = db.CreateMessage(msg, canal, o.database)
 	if errs != nil {
-		return nil, nil, err
+		log.Printf("An error occured, unable to store received message in the database")
+		return nil, errs
 	}
 
-	return parser.ComposeBroadcastMessage(query), []byte(canal), nil
+	msgAndChan := []lib.MessageAndChannel{{
+		Message: parser.ComposeBroadcastMessage(query),
+		Channel: []byte(canal),
+	}}
+
+	log.Printf("Sucessfully created poll %s", poll.Name)
+
+	return msgAndChan, nil
 }
 
-func (o *Organizer) handleUpdateProperties(msg message.Message, canal string, query message.Query) (message, channel []byte, err error) {
-	return parser.ComposeBroadcastMessage(query), []byte(canal), db.CreateMessage(msg, canal, o.database)
+// handleUpdateProperties is the function to handle a received message requesting a change of some properties of a LAO.
+// It is called by the function handlePublish.
+// The received message had the object field set to "lao" and action field to "update_properties"
+// It will store the received message in the database, and send the change request to every subscriber of this LAO,
+// waiting for Witnesse's validation to make the update.
+func (o *Organizer) handleUpdateProperties(msg message.Message, canal string, query message.Query) (msgAndChannel []lib.MessageAndChannel, err error) {
+	// TODO need some improvements
+	msgAndChan := []lib.MessageAndChannel{{
+		Message: parser.ComposeBroadcastMessage(query),
+		Channel: []byte(canal),
+	}}
+	return msgAndChan, db.CreateMessage(msg, canal, o.database)
```

Comment:
well actually it's fully fonctionnal... Just needs to have an if statement to check that if SigThreshold == 0 then directly make the stateUpdate but otherwise is ok for me

Reply:
For me we can even remove the TODO... one thing that **could** be done if check if the message's sender is legitimate to request changes on a LAO. (by comparing the sender's public key with the BE's public key for example)...